content/6_Advanced/DP_SOS.mdx

```diff
@@ -77,7 +77,10 @@ vector<int> a(1 << n);
 vector<int> sos(1 << n);
 for (int i = 0; i < (1 << n); i++) {
 	// iterate over all subsets of i directly
-	for (int j = (i - 1) & i; j >= 0; j = (j - 1) & i) { sos[i] += a[j]; }
+	for (int j = (i - 1) & i;; j = (j - 1) & i) {
```

Comment:
i might be tweaking but would `j > 0` as a loop condition do the same thing here

Reply:
I think the previous one was right it doesn't need any changes.

Reply:
envy said it was wrong so he’s prolly right

Reply:
j > 0 would not work because we want the loop to run when j = 0

The previous condition is wrong because when j = 0, the value will be updated to be (0 - 1) & i which is equal to i. Then when the loop condition is being checked, it would pass since i >= 0.

Reply:
oh lolz

Reply:
Why not add the condition `j != 0` in the loop only

Reply:
read his comment 💀

Reply:
my bad 😅

Reply:
great observation @Keys9873